NitroxClient/Communication/Packets/Processors/CyclopsToggleEngineStateProcessor.cs

```diff
@@ -0,0 +1,43 @@
+using NitroxClient.Communication.Packets.Processors.Abstract;
+using NitroxClient.GameLogic.Helper;
+using NitroxModel.DataStructures.Util;
+using NitroxModel.Helper;
+using NitroxModel.Packets;
+using System;
+using UnityEngine;
+
+namespace NitroxClient.Communication.Packets.Processors
+{
+    public class CyclopsToggleEngineStateProcessor : ClientPacketProcessor<CyclopsToggleEngineState>
+    {
+        private PacketSender packetSender;
+
+        public CyclopsToggleEngineStateProcessor(PacketSender packetSender)
+        {
+            this.packetSender = packetSender;
+        }
+
+        public override void Process(CyclopsToggleEngineState enginePacket)
+        {
+            Optional<GameObject> opCyclops = GuidHelper.GetObjectFrom(enginePacket.Guid);
+
+            if (opCyclops.IsPresent())
+            {
+                CyclopsEngineChangeState engineState = opCyclops.Get().GetComponentInChildren<CyclopsEngineChangeState>();
+
+                if (enginePacket.IsOn == engineState.motorMode.engineOn)
+                {
+                    using (packetSender.Suppress<CyclopsEngineChangeState>())
+                    {
+                        engineState.ReflectionSet("startEngine", true);
```

Comment:
Is this necessary? `OnClick` will set it to true anyway when the Cyclops if turned off. And when the Cylcops is turned on, this variable is not even read.

Reply:
Yes it is necessary. Prefix() will be set __state to !motor-state and no Packet will send back...

Reply:
You're already encapsulating this method in a suppressor.